modules/vault/main.tf

```diff
@@ -44,9 +41,21 @@ resource "azurerm_key_vault_access_policy" "identity" {
   tenant_id    = data.azurerm_client_config.current.tenant_id
   object_id    = var.identity_object_id
 
-  key_permissions     = ["Get"]
-  secret_permissions  = ["Get", "Delete", "List", "Purge", "Recover", "Restore", "Set"]
-  storage_permissions = ["Get"]
+  key_permissions     = ["Create", "Decrypt", "Encrypt", "Get", "List"]
```

Comment:
Just double checking, we added "Create", "Decrypt", "Encrypt" on line 32 and that wasn't enough to allow the deploying principal to create keys but after we added "Create", "Decrypt", "Encrypt" here, it was?

I do also find it weird that we need to give these to cluster identity to create keys in the key vault..

Reply:
After some experimentation, looks like there is no clearly documented way to manually send requests as the user-assigned managed identity even with access to the cluster